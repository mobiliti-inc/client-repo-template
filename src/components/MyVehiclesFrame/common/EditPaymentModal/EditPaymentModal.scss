@import '~variables';

.modal-title {
	font-size: 24px;
	font-family: $futura;
	font-weight: bold;
	color: $darker-blue;
}

.modal-ctn {
	white-space: normal;
	a:last-child {
		> div {
			border: none;
		}
	}
}

.modal-content {
	cursor: pointer;
	border-bottom: 1px solid rgba(29, 49, 85, 0.2);
	p:first-child {
		font-size: 18px;
		font-family: $futura;
		font-weight: bold;
		line-height: 31px;
		margin-bottom: 0;
		color: $darker-blue;
	}
	p:last-child {
		font-size: 16px;
		font-family: $hanken-sans;
		opacity: 0.8;
		line-height: 19px;
		margin: 0 auto 25px auto;
		color: $darker-blue;
	}
}

