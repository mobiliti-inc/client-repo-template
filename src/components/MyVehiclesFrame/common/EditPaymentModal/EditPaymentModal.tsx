import Modal from '../../../Modal/Modal';
import Link from '../../../Link/Link';
import styles from './EditPaymentModal.scss';

const EditPaymentModal = ({
	modalVisible,
	scrolling,
	closeModal,
	editSubscriptionRoute,
	onSelectSwap,
	onSelectReturnVehicle
}) => (
	<Modal
		plain
		visible={modalVisible}
		overlay="gray"
		scrolling={scrolling}
		onIconClose={closeModal}
	>
		<p styleName="modal-title">Edit Payment</p>
		<div styleName="modal-ctn">
			<Link to={editSubscriptionRoute}>
				<div styleName="modal-content">
					<p>Edit Subscription</p>
					<p>Change your mileage options or payment methods today.</p>
				</div>
			</Link>

			<div styleName="modal-content" role="presentation" onClick={onSelectReturnVehicle}>
				<p>Return Vehicle</p>
				<p>Don&#39;t need your vehicle anymore? Return it hassle free.</p>
			</div>

			<div styleName="modal-content" id="swap" onClick={onSelectSwap} role="presentation">
				<p>Swap Vehicles</p>
				<p>Ready for something new? We can help you get into a new vehicle.</p>
			</div>
		</div>
	</Modal>
);

EditPaymentModal.propTypes = {
	modalVisible: PropTypes.bool.isRequired,
	scrolling: PropTypes.bool.isRequired,
	closeModal: PropTypes.func.isRequired,
	editSubscriptionRoute: PropTypes.string.isRequired,
	onSelectSwap: PropTypes.func.isRequired,
	onSelectReturnVehicle: PropTypes.func.isRequired,
};

export default CSSModules(EditPaymentModal, styles, { allowMultiple: true });
