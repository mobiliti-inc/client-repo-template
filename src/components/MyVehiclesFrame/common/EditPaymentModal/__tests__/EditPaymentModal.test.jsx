import EditPaymentModal from '../EditPaymentModal';

describe('EditPaymentModal', () => {
	const props = {
		modalVisible: false,
		scrolling: false,
		closeModal: jest.fn(),
		editSubscriptionRoute: '/',
		onSelectSwap: jest.fn(),
		onSelectReturnVehicle: jest.fn(),
	};
	const component = <EditPaymentModal {...props} />;

	it('renders', () => {
		expect(component).toMatchSnapshot();
	});
});
