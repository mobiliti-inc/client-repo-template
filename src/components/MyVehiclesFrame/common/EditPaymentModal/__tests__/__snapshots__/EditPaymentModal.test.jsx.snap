// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditPaymentModal renders 1`] = `
<EditPaymentModal
  closeModal={[MockFunction]}
  editSubscriptionRoute="/"
  modalVisible={false}
  onSelectReturnVehicle={[MockFunction]}
  onSelectSwap={[MockFunction]}
  scrolling={false}
/>
`;
